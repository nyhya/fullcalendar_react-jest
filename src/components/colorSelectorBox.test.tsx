import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import { shallow } from 'enzyme';
import ColorSelectorBox from 'components/colorSelectorBox';

describe('<ColorSelectorBox/>', () => {
	const myMock = jest.fn();
	const { container } = render(
		<ColorSelectorBox color="#F44336" click={myMock} />,
	);
	test('renders component correctly', () => {
		expect(container).toBeInTheDocument();
		expect(container).toMatchSnapshot();
	});

	test('renders', () => {
		render(<ColorSelectorBox color="#F44336" click={myMock} />);
		const colorBox = screen.getByTestId('colorBox');
		fireEvent.click(colorBox);
	});
});
