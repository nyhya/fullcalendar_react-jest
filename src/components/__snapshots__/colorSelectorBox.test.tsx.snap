// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ColorSelectorBox/> renders component correctly 1`] = `
<div>
  <div
    class="sc-bczRLJ grVmKH"
    color="#F44336"
  >
    <div
      class="colorBox"
      data-testid="colorBox"
    />
  </div>
</div>
`;
